### ISSUE TITLE
Can't launch/connect with Flash Player in Ubuntu

### ISSUE THREAD

**Issue by diegosolo** [vshaxe/vshaxe-debugadapter#10]
Hi, I'm trying to properly install the flash debugger in Xubuntu 16.04 with no luck. I wonder if it's just me doing things the wrong way, but there's just too little information around about this specific subject. This is what I've done so far:

- Install Haxe 3.4
	sudo add-apt-repository ppa:haxe/releases -y
	sudo apt-get update
	sudo apt-get install haxe -y
	mkdir ~/.haxelib-repository && haxelib setup ~/.haxelib-repository

- Install Java JDK
	sudo apt-get install default-jdk
	(path: /usr/lib/jvm/default-java/)

- Set JAVA_HOME enviroment variable
	Open /etc/environment
	Add the line: JAVA_HOME="/usr/lib/jvm/default-java/"
	Reload variables: ource /etc/environment
	Check: echo $JAVA_HOME

- Install flixel
	haxelib install flixel
	haxelib install flixel-tools
	haxelib run lime setup
	lime setup android

- Install g++
	sudo apt-get install g++

- Install Flash Player Debugger
	Download https://fpdownload.macromedia.com/pub/flashplayer/updaters/25/flash_player_sa_linux_debug.x86_64.tar.gz
	Decompress and copy flashplayerdebugger to /usr/bin
	Add an item to the system menu
	Associate swf files

- Download and install Visual Studio from https://code.visualstudio.com/Download

- Install Haxe Extension Pack (vshaxe) from the package manager

- Configure Visual Studio (http://haxeflixel.com/documentation/visual-studio-code/)
	navigate to File -> Preferences -> Keyboard Shortcuts and change keybindings.json on the right with the following:
		// Place your key bindings in this file to overwrite the defaults
		[
			{ "key": "ctrl+b", "command": "workbench.action.tasks.runTask" }
		]

After doing this, I can open any swf file by double clicking it. Yet, when I run "flash build + launch" from Visual Studio the compilation runs but then nothing happens; it just stays waiting. I tried opening the movie with a double click while the debugger is waiting with no success (the window opens but the debugger keeps waiting). I'd appreciate some help.

Thanks!

**Comment by Gama11** [comment]
Are you on the latest version of vshaxe-debugadapter (1.1.1)? The most recent patch release in particular solved an issue on Linux that yours sounds very similar too.

Does the .swf file open when just running the flash build task, without debugging?

Is there any interesting output in the Debug Console if you add the following to your `launch.json`'s Launch Config?

```json
"receiveAdapterOutput": true
```

**Comment by diegosolo** [comment]
Hi, Thanks for the prompt answer.

- It is the last version (1.1.1).
- No, it doesn't. Building goes allright (I can check it by manually opening the swf and seeing the changes); the problem is when trying to launch the Flash Player with or without previous compilation.
- Yes! Very interesting:

DebugAdapter: send to debugger cli: run _[path-to-movie]_.swf
running _[path-to-movie]_.swf
DebugAdapter: Launch: [Adobe fdb (Flash Player Debugger) [build 23201],Copyright (c) 2004-2007 Adobe, Inc. All rights reserved.]
DebugAdapter: {"command":"configurationDone","type":"request","seq":3}
DebugAdapter: sendResponse: {
	seq : 0, 
	type : response, 
	request_seq : 3, 
	command : configurationDone, 
	success : true
}
DebugAdapter: {"command":"threads","type":"request","seq":4}
DebugAdapter: sendResponse: {
	seq : 0, 
	type : response, 
	request_seq : 4, 
	command : threads, 
	success : true, 
	body : {
		threads : [{
				id : 1, 
				name : thread 1
			}]
	}
}
DebugAdapter: Launch: [Attempting to launch and connect to Player using URL,_[path-to-movie]_.swf,File not found: flashplayer]

Note: I replaced the file name and path by _[path-to-movie]_.

Moreover, the file I downloaded from adobe was named "flashplayerdebugger" (I placed it in the /usr/bin/ folder). So, I made a copy and named it just "flashplayer". The debugger launches the Flash Player and I can see the variables when I pause the game.

It's solved for me now but you may take this into account for a future update. The application "flashplayerdebugger" was properly associated to swf files (I could open the files with a double-click). Anything else you may need just ask. Thanks!

**Comment by diegosolo** [comment]
Oh, by the way. In linux you can open a file with its associated program by using the [xdg-open](https://linux.die.net/man/1/xdg-open) command, just in case it's useful.

**Comment by Gama11** [comment]
@ibilon Anything we could do about this?

**Comment by ibilon** [comment]
It's `fdb.jar` which launches the swf, unless we recompile it we can't do much.

**Comment by diegosolo** [comment]
Ok, I made a test and it doesn't matter if swf files are associated to the player or not. Look like `fdb.jar` just launches the application named `flashplayer`. It could at least be noted in [GitHub's description page](https://github.com/vshaxe/vshaxe-debugadapter#haxe-debug) and in the [VSCode description page](https://marketplace.visualstudio.com/items?itemName=vshaxe.haxe-debug#haxe-debug), so others may not be at a lost anymore. It can be very frustrating and I would have assumed it just didn't work if I were able to use Windows and FlashDevelop.

Thanks for the help!
